test/remittance.js

```diff
@@ -4,7 +4,371 @@ const Remittance = artifacts.require("./Remittance.sol");
+
+        // Check contract's changed ETH balance
+        const contractEthBalance = toBN(await web3.eth.getBalance(remittance.address));
+
+        assert.strictEqual(contractEthBalance.toString(10), "2");
+
+        await truffleAssert.eventEmitted(txObj, 'RemittanceFundsCreated', (ev) => {
+
+            return  ev.hash === rehash &&
+                    ev.funder === funder &&
+                    ev.broker === broker &&
+                    ev.amount.toString(10) === "2" &&
+                    ev.expiryDate.toString(10) === expiryDate.toString(10);
+        }, 'RemittanceFundsCreated event is emitted');
+
+        const remittanceInstance = await remittance.remittances(rehash);
+
+        assert.strictEqual(remittanceInstance.funder, funder);
+        assert.strictEqual(remittanceInstance.broker, broker);
+        assert.strictEqual(remittanceInstance.fundsOwed.toString(10), "2");
+        assert.strictEqual(remittanceInstance.expiryDate.toString(10), expiryDate.toString(10));
+        assert.isNotFalse(remittanceInstance.valid);
+    });
+
+    it("Broker releases funds to their account", async () => {
+
+        await remittance.create(rehash, broker, expiryDate, {from: funder, value: 2});
+        const initContractEthBalance = toBN(await web3.eth.getBalance(remittance.address));
+        const initBrokerEthBalance = toBN(await web3.eth.getBalance(broker));
+
+        const txObj = await remittance.release(hash, {from: broker});
+
+        await truffleAssert.eventEmitted(txObj, 'RemittanceFundsReleased', (ev) => {
+            return  ev.hash === rehash &&
+                    ev.broker === broker &&
+                    ev.amount.toString(10) === "2";
+        }, 'RemittanceFundsReleased event is emitted');
+
+        const remittanceInstance = await remittance.remittances(rehash);
+
+        assert.strictEqual(remittanceInstance.fundsOwed.toString(10), "0");
+        assert.strictEqual(remittanceInstance.expiryDate.toString(10), "0");
+
+        // Check the remittance amount has been taken from the contract eth balance
+        const contractEthBalance = toBN(await web3.eth.getBalance(remittance.address));
+        const expectedContractEthBalance = initContractEthBalance.sub(toBN(2)).toString(10);
+        assert.strictEqual(contractEthBalance.toString(10), expectedContractEthBalance);
+
+        // Check the remittance amount has been sent to the broker eth balance
+        const brokerEthBalance = toBN(await web3.eth.getBalance(broker));
+        const cost = await getGasCost(txObj);
+        const expectedBrokerEthBalance = initBrokerEthBalance.add(toBN(2)).sub(toBN(cost)).toString(10);
+        assert.equal(brokerEthBalance.toString(10), expectedBrokerEthBalance);
+
+    });
+
+    it("Funder reclaims funds from an expired remittance", async () => {
+
+        //Create a remittance with an expired date
+        await remittance.create(rehash, broker, expiredDate, {from: funder, value: 2});
```

Comment:
Mmh, I can see users making the mistake of creating one with an expired date...
Instead, use `advanceTime` of Ganache.

Reply:
Oh. Yes. I should perhaps also check that the expiry is at a minimum of 24 hours in the future.